Patch to address 20638369.  This patch is Solaris specific and not
suitable for upstream contribution.

It also addresses similar problem with xgettext.

--- Django-4.2/django/core/management/commands/compilemessages.py
+++ Django-4.2/django/core/management/commands/compilemessages.py
@@ -32,7 +32,7 @@ class Command(BaseCommand):
 
     requires_system_checks = []
 
-    program = "msgfmt"
+    program = "/usr/gnu/bin/msgfmt"
     program_options = ["--check-format"]
 
     def add_arguments(self, parser):
--- Django-4.2/django/core/management/commands/makemessages.py
+++ Django-4.2/django/core/management/commands/makemessages.py
@@ -424,7 +424,7 @@ class Command(BaseCommand):
         if locales:
             check_programs("msguniq", "msgmerge", "msgattrib")
 
-        check_programs("xgettext")
+        check_programs("/usr/gnu/bin/xgettext")
 
         try:
             potfiles = self.build_potfiles()
@@ -480,7 +480,7 @@ class Command(BaseCommand):
         # Gettext tools will output system-encoded bytestrings instead of UTF-8,
         # when looking up the version. It's especially a problem on Windows.
         out, err, status = popen_wrapper(
-            ["xgettext", "--version"],
+            ["/usr/gnu/bin/xgettext", "--version"],
             stdout_encoding=DEFAULT_LOCALE_ENCODING,
         )
         m = re.search(r"(\d+)\.(\d+)\.?(\d+)?", out)
@@ -637,7 +637,7 @@ class Command(BaseCommand):
         if self.domain == "djangojs":
             is_templatized = build_file.is_templatized
             args = [
-                "xgettext",
+                "/usr/gnu/bin/xgettext",
                 "-d",
                 self.domain,
                 "--language=%s" % ("C" if is_templatized else "JavaScript",),
@@ -650,7 +650,7 @@ class Command(BaseCommand):
             ]
         elif self.domain == "django":
             args = [
-                "xgettext",
+                "/usr/gnu/bin/xgettext",
                 "-d",
                 self.domain,
                 "--language=Python",
